### PR TITLE
Enable migration resource in v12 with credentials logic

### PR DESCRIPTION
**Pull request by rossf7**
Fixes giantswarm/giantswarm#3406

### CODE REVIEW

File: service/controller/v12/resource/migration/resource.go [hunk 1]
```diff
@@ -61,30 +57,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("settings all zones to %q", zone))
-
-		customObject.Spec.AWS.HostedZones.API.Name = zone
-		customObject.Spec.AWS.HostedZones.Etcd.Name = zone
-		customObject.Spec.AWS.HostedZones.Ingress.Name = zone
-	}
-
-	if reflect.DeepEqual(customObject.Spec, oldSpec) {
+	// If credential secret is set we have nothing to migrate.
+	if customObject.Spec.AWS.CredentialSecret.Name != "" {
```
Comment (rossf7) [inline service/controller/v12/resource/migration/resource.go]: Enables migration resource with logic from v11. This lets us upgrade directly to v12.

@TheoBrigitte Does this look OK?

**Comment by xh3b4sd** [comment]
Based on https://github.com/giantswarm/giantswarm/issues/3406#issuecomment-397261635 I think we should also add it to `v13` and keep it there until we are completely safe and until actual data suggests so.

**Comment by rossf7** [comment]
@xh3b4sd Yes just thought that too. I've added to v13.

File: service/controller/v12/resource/migration/resource.go (hunk 1 above)
Comment (TheoBrigitte) [reply service/controller/v12/resource/migration/resource.go]: Looks all fine

Comment (rossf7) [reply service/controller/v12/resource/migration/resource.go]: Cool thanks! 👍

**Comment by rossf7** [comment]
Fishing for an approval here.

**Pull request merged**